apps/app/src/components/Common/Dropdown/PageItemControl.tsx

```diff
@@ -245,12 +245,51 @@ const PageItemControlDropdownMenu = React.memo((props: DropdownMenuProps): JSX.E
     );
   }
 
+  // Conditional modifiers
+  // To prevent flickering. only happened when `right` is true and persist props should be enabled
+  let modifiers = {};
+  if (alignRight) {
+    modifiers = {
+      applyStyle: {
+        enabled: true,
+      },
+      computeStyle: {
+        enabled: true,
+        fn: (data) => {
+          const popperRect = data.offsets.popper;
+          // Calculate transform styles
+          const newTransform = `translate3d(${popperRect.left - window.innerWidth + popperRect.width}px, ${popperRect.top}px, 0px)`;
+          const styles = {
+            transform: newTransform,
+            top: '0px',
+            right: '0px',
+            willChange: 'transform',
+          };
+          data.styles = styles;
+          return data;
+        },
+      },
+      preventOverflow: {
+        boundariesElement: 'viewport',
+      },
+    };
+  }
+  else {
+    modifiers = {
+      preventOverflow: {
+        boundariesElement: 'viewport',
+      },
+    };
+  }
```

Comment:
Great job 👍 

I think it is better to commonize the `modifiers`.
Would you move this as `modifiersForRightAlign` under `@growi/ui`?